modtools/transform-unit.lua

```diff
@@ -108,7 +108,12 @@ else
  end
 
  if not caste then
-  error 'Invalid caste.'
+  local rand_caste = math.random( 0, math.abs(#race.caste-1) )
+  if not race.caste[rand_caste] then
+    error 'Creature has no castes or non have been found'
+  else
```

Comment:
No need to have an else clause if the if clause throws an error. Just replace the else with an end and let the assignment happen unconditionally.